library/ccm.c

```diff
@@ -99,7 +101,14 @@ int mbedtls_ccm_setkey( mbedtls_ccm_context *ctx,
         return( ret );
     }
 
-    return( ret );
+    if( keybits_dup == keybits && key_dup == key )
+    {
+        return( ret );
+    }
+
+    // In case of a FI - clear the context
+    mbedtls_cipher_free( &ctx->cipher_ctx );
```

Comment:
@AndrzejKurek , i think we should set the pointer to  NULL, otherwise we will have problem with  line 93

Reply:
This is not a pointer, but a structure inside a structure.  We can't set it to NULL. We could clear it to the state it was before, by calling the cipher_free, which we're doing.